remotes/docker/pusher.go

```diff
@@ -375,22 +377,20 @@ func (pw *pushWriter) Write(p []byte) (n int, err error) {
 			if !ok {
 				return 0, io.ErrClosedPipe
 			}
-			pw.pipe.CloseWithError(content.ErrReset)
+			pw.error(content.ErrReset)
```

Comment:
I would think since `*pushWriter` owns the write side and can already control the error returned, it makes the most sense here to close the write side so the reader gets the error. What is the expected error here the reader should get.

Reply:
I think that should be good - I'll find an error message that makes sense for this.

The reason I did it like this is so that multiple calls to the `Write` function would return `ErrReset` - we get a form of persistence by closing the `Reader`. Without it, we'd get 1 `ErrReset` and then `io.PipeClosedError` - I think that's fine though.